modules/guides/src/GuidesStorage.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Drupal\guides;
+
+/**
+ * Copyright (C) 2019 PRONOVIX GROUP BVBA.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+ * USA.
+ */
+
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Url;
+use Drupal\Core\Link;
+use Drupal\Core\Site\Settings;
+
+/**
+ * Class GuidesStorage.
+ */
+class GuidesStorage implements GuidesInterface {
+
+  /**
+   * The settings of the site.
+   *
+   * @var \Drupal\Core\Site\Settings
+   */
+  protected $settings;
+
+  /**
+   * The cache backend service.
+   *
+   * @var \Drupal\Core\Cache\CacheBackendInterface
+   */
+  protected $cacheBackend;
+
+  /**
+   * GuidesStorage constructor.
+   *
+   * @param \Drupal\Core\Site\Settings $settings
+   *   The settings of the site.
+   */
+  public function __construct(Settings $settings, CacheBackendInterface $cache_backend) {
+    $this->settings = $settings;
+    $this->cacheBackend = $cache_backend;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDirectory(): string {
+    return DRUPAL_ROOT . ($this->settings->get('guides_dir') ?? '/guides');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDirectories(): array {
```

Comment:
If I am not mistaken I have managed to remove all usages of this method outside of this class. If we can get rid of this method then please remove it you can make it `private`.

Reply:
I couldn't make it private, because I have to give the directory name in the cid where I try to load the files from cache, so I have to use it outside of this class.